src/messaging/ExchangeContext.cpp

```diff
@@ -84,6 +85,12 @@ CHIP_ERROR ExchangeContext::SendMessage(Protocols::Id protocolId, uint8_t msgTyp
 
     VerifyOrReturnError(mExchangeMgr != nullptr, CHIP_ERROR_INTERNAL);
 
+    // TODO(#6220): We add this check to make platforms that does not implemented full platform (i.e. Android) can compile, and does
+    // not means that those platforms do not need this.
+#if CONFIG_DEVICE_LAYER
+    DeviceLayer::CHIPPlatformLock platformLock;
```

Comment:
I think we should move this lock to SendMessageImpl since not all messages are sent from this interface, such as MSCP etc...

Reply:
Then I would suggest add another lock at SecureSeesionManager::SendMessage as well.
Since ExchangeContext::SendMessage and other ExchangeContext related operations can be take at different tasks / threads, we should have a lock here.

Reply:
@yufengwangca Do all messages go through SecureSessionMgr?  I would have thought unencrypted ones do not.

Reply:
The exchange is designed to paralleling the send/receive operations, so that we can have multiple "conversations" simultaneously, and each separate conversation is represented as an exchange, different protocols can use different exchange to conduct its own conversation independent from other exchange.

If we put lock inside  ExchangeContext::SendMessage, then we serialize the send operations among different exchanges. If the root cause of this crash is we try to clear the message before it is cached, I think adding the lock within SecureSessionMgr::SendMessage should be able to solve this issue.